### ISSUE TITLE
Overhaul README.md in this repository

### ISSUE THREAD

**Issue by sxa** [adoptium/temurin-build#3860]
There have been very few updates to the top level README.md in the temurin-build and ci-jenkins-pipelines repositories, and so we should look at them and see if they are "fit for purpose" and whether the information on here that we present to our user base is (a) useful (b) accurate and (c) fit for purpose.

I've started with an overhaul of the ci-jenkins-pipelines README.md in https://github.com/adoptium/ci-jenkins-pipelines/pull/1048 but here are some discussion points for the one in here:

- We have multiple ways of setting up a build environment - what do we want to recommend and promote?
  - Run the playbooks on your local machine
  - List the prereqs that are required (Maybe complex as it's different on each platform?)
  - Use the `-D` option on makejdk-any-platform.sh to automatically build in a new container image to avoid discouraging podman users (Also, why do we have this section yet also reference the dockerfile generator in the "build natively" section? Is this likely to be confusing to first time users?) NOTE: This is converted to `-D` in https://github.com/adoptium/temurin-build/pull/3796/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5
  - Duplicating the "man page" for makejdk-any-platform.sh in the README.md seems unnecessary - we should only have this information in one place ... But which location is right? Is the "man page" format still useful?
  - Is the metadata section useful? Should it be in this repository's README or the ci-jenkins-pipelines one? My gut feel is that it should probably be here, but maybe collapsed (as I did in my pipelines PR for now) or moved elsewhere as it's perhaps not "interesting" particularly now that we have the SBoM generation.
  - We should perhaps shout more about SBoM and reproducibility options in the README.md
  - Should SmokeTesting.md and Testing.md be moved somewhere else instead of the top level directory? Maybe Testing.md should be part of CONTRIBUTING.md and the smoke test doc to `test/functional/buildAndPackage/README.md`?

**Comment by judovana** [comment]
One friendly note. The man page is good. The best of its existence  is the usage during --help for build: https://github.com/adoptium/temurin-build/blob/8646220187f9e64087249085e519a9d3d18ae97e/sbin/common/config_init.sh#L320  . Maybe the `.1` file can be linked  to its generated html version?

**Comment by sxa** [comment]
>  Maybe the .1 file can be linked to its generated html version?

Yeah I just want to make sure it's only in one place to avoid duplication/fragmentation (Currently it's in the README and [the separate man page](https://github.com/adoptium/temurin-build/blob/master/makejdk-any-platform.1)). I've seen quite a few PRs that have only updated one of them. I'd probably only have one of them and link one to there other, although I'm unsure which way round is best to cover both people using the tool running `-h` and people looking at the README (The `.1` man pages aren't too easy to read via the web). All opinions welcome though :-)

**Comment by judovana** [comment]
Sure thing to get rid of duplication. Linking the .1 file from readme.md seems like initial step to do. When I was searching for "man2html " online, I was hoping to find something like http://raw.githack.com , just with additional man page convertion.   

Maybe it can be tan be taken form different angle. There seems to be some efforts to have man pages in markup. Maybe if there is this man page in markup, it will be easy to provide to man, to html view and also as plain text view.
